my_library/models/library_book.py

```diff
@@ -1,15 +1,111 @@
+from odoo.exceptions import ValidationError
 
 
 class LibraryBook(models.Model):
     _name = 'library.book'
+    _description = 'Library Book'
+
+    _order = 'date_release desc, name'
+    _rec_name = 'short_name'
 
     name = fields.Char(string='Title', required=True)
+    short_name = fields.Char(string='Short Title', required=True)
+    notes = fields.Text(string='Internal Notes')
+    state = fields.Selection(
+        selection=[('draft', 'Not Available'),
+                   ('available', 'Available'),
+                   ('lost', 'Lost')
+                   ],
+        string='State'
+    )
+    description = fields.Html(string='Description')
+    cover = fields.Binary(string='Book Cover')
+    out_of_print = fields.Boolean(string='Out of Print?')
     date_release = fields.Date(string='Release Date')
-    author_ids = fields.Many2many(
+    date_update = fields.Datetime(string='Last Update', copy=False)
+    pages = fields.Integer(
+        string='Number of pages',
+        groups='base.group_user',
+        states={'lost': [('readonly', True)]},
+        help='Total book page count', company_dependent=False
+    )
+    reader_rating = fields.Float(string='Reader Average Rating', digits=(14, 4))
+    cost_price = fields.Float(string='Book Cost', digits=dp.get_precision('Book Price'))
+    currency_id = fields.Many2one(comodel_name='res.currency', string='Currency')
+    retail_price = fields.Monetary(string='Retail Price', currency_field='currency_id')
+    publisher_id = fields.Many2one(
         comodel_name='res.partner',
-        relation='book_partner',
-        column1='book_id',
-        column2='partner_id',
-        string='Authors'
+        string='Publisher',
+        ondelete='set null',
+        context={},
+        domain=[]
     )
+    publisher_city = fields.Char(string='Publisher City', related='publisher_id.city', readonly=True)
+    category_id = fields.Many2one(comodel_name='library.book.category')
+    _sql_constraints = [
+        ('name_uniq', 'UNIQUE (name)', 'Book title must be unique.'),
+        ('positive_page', 'CHECK(pages>0)', 'No of pages must be positive')
+    ]
+    age_days = fields.Float(
+        string='Days Since Release',
+        compute='_compute_days',
+        inverse='_inverse_age',
+        search='_search_age',
+        store=False,  # optional
+        compute_sudo=True  # optional
+    )
+
+    ref_doc_id = fields.Reference(selection='_referencable_models', string='Reference Document')
+
+    def name_get(self):
+        result = []
+        for record in self:
+            rec_name = "%s %s" % (record.name, "(%s)" % record.date_release if record.date_release else '')
+            result.append((record.id, rec_name))
+        return result
+
+    @api.constrains('date_release')
+    def _check_release_date(self):
+        for record in self:
+            if record.date_release > fields.Date.today():
+                raise ValidationError('Release date must be in the past')
```

Comment:
from odoo import _

raise ValidationError(_('Release date must be in the past'))